src/npcmove.cpp

```diff
@@ -376,6 +376,22 @@ void npc::assess_danger()
         max_range = *confident_range_cache;
     }
     Character &player_character = get_player_character();
+    const bool self_defense_only = rules.engagement == combat_engagement::NO_MOVE ||
+                                   rules.engagement == combat_engagement::NONE;
+    const bool no_fighting = rules.has_flag( ally_rule::forbid_engage );
+    const bool must_retreat = rules.has_flag( ally_rule::follow_close ) &&
+                              !too_close( pos(), player_character.pos(), follow_distance() );
```

Comment:
I believe this makes NPCs with 'stick close to me, no matter what' rule ignore enemies if player tells them to guard a position and then walks away.